[Soh Xin Wei] iP

xiinweii98 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

xiinweii98 committed:
Level 1

xiinweii98 committed:
Level 2

xiinweii98 committed:
Level 3

xiinweii98 committed:
Level 4

xiinweii98 committed:
Automated text ui testing

xiinweii98 committed:
Level 5

xiinweii98 committed:
Level 6

xiinweii98 committed:
Level 7 of Duke individual project

Added level 7 of the project - saving tasks list to hard drive and retrieving it whenever Duke runs.

xiinweii98 committed:
Added Level 8 to Duke individual project

Duke now takes in date and store it as a LocalDate data instead of a String.

xiinweii98 committed:
Merge branch 'branch-Level-7'

xiinweii98 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

xiinweii98 committed:
Reorganised code to follow OOP principles

Created 4 classes to deal with general usage:
Ui - to deal with interactions with the user (printing messages and reading inputs)
Storage - to deal with loading/writing tasks from/to file in hard drive
Parser - to deal with different commands input from user
TaskList - to deal with organising the tasks (add/delete/markDone etc)

Created another 6 classes with a superclass Command and 5 subclasses
AddCommand - to deal with adding different tasks
DeleteCommand - to deal with deleting a certain task
DoneCommand - to deal with marking a certain task as done
ExitCommand - to deal with exiting the program
ListCommand - to deal with displaying the list of tasks at hand

Some exceptions are not thrown/handled properly yet. There are some cases where the program will be unable to deal with.
Example: program does not deal with deletion/marking of a task that does not exist

xiinweii98 committed:
Tried packaging

xiinweii98 committed:
Added some JUnit tests

Added test cases for Task class only. To add more test cases for other classes.

xiinweii98 committed:
Added JavaDoc for Classes and methods.

xiinweii98 committed:
Minimal changes made for A-CodingStandard.

Could not find any other mismatches with original code and recommended coding standards.

xiinweii98 committed:
Added Level-9, a way to search for tasks using a keyword.

xiinweii98 committed:
Merge tag 'branch-A-JavaDoc'

xiinweii98 committed:
Merge tag 'branch-A-CodingStandard'

xiinweii98 committed:
Merge tag 'branch-Level-9'

stein414 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    @Override
+    public String getStatus() {
```
Perhaps add a JavaDoc for this method? Similarly for other public methods.

stein414 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,42 @@
+    public static Command parseCommand(String fullCommand) throws DukeException {
+        String[] first = fullCommand.split(" ", 2);
+        switch (first[0]) {
+        case ("bye"):
```
Perhaps the `()` are redundant and could be removed?

stein414 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,110 @@
+            }
+        }
+
+        this.file = savedTasks;
```
Perhaps the `this` can be removed since file is not being shadowed. Similarly for other places.

stein414 commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,110 @@
+            String current = sc.nextLine();
+            String[] splits = current.split(" ", 3);
+            switch (splits[0]) {
+            case "T": {
```
Maybe remove the extra {} for the case block?

stein414 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,72 @@
+    }
+
+    /**
+     * Add a task.
```
Perhaps `Adds` would fit better here? Similarly for other places

stein414 reviewed (commented):
Overall it looks good for the current level the commit is for. It just needs a little cleaning up in terms of coding standard violations and some logic could be rearrange to make the code flow a little better.

xiinweii98 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

# Conflicts:
#	text-ui-test/runtest.sh

xiinweii98 committed:
Merge tag 'branch-A-Gradle'

xiinweii98 committed:
Branch A-CheckStyle

xiinweii98 committed:
Merge tag 'branch-A-CheckStyle'

marc-97 commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,42 @@
+     * @throws DukeException If user input an invalid command.
+     */
+    public static Command parseCommand(String fullCommand) throws DukeException {
+        String[] first = fullCommand.split(" ", 2);
+        switch (first[0]) {
```
Perhaps use a more descriptive name for "first"

marc-97 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Override
+    public String getStatus() {
+        return "[D]" + super.getStatus() + " (by: " + this.by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
Perhaps you could add a line break to improve readability

marc-97 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,58 @@
+     * @return The status of the Task.
+     */
+    public String getStatus() {
+        return "[" + (this.isDone ? "\u2713" : "\u2718") + "] " + this.name;
```
Maybe declare a constant for "\u2713" & "\u2718"

marc-97 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,156 @@
+     * Prints the list of tasks according to the input.
+     * @param tasks The list of tasks to print.
+     */
+    public void printSomeTasks(ArrayList<Task> tasks) {
```
Maybe change the method name as it is currently printing all the items in tasks

marc-97 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,156 @@
+     * Prints the number of tasks.
+     * @param taskList The TaskList that is being kept tracked of.
+     */
+    public void printNumTasks(TaskList taskList) {
```
Perhaps you could change the variable name from taskList to tasks. Same for the other methods

marc-97 reviewed (commented):
Overall, code is good and easy to read 👍 . Could package all the classes as recommended by the java coding standard.

xiinweii98 committed:
Added GUI to Duke.

Follow steps in guide. Yet to customize. Yet to optimize.

xiinweii98 committed:
Merge tag 'branch-Level-10'

maxxng commented on src/main/java/AddCommand.java:
```diff
@@ -1,12 +1,28 @@
     }
 
+    /**
+     * Adds the task into the TaskList and update the file in hard drive respectively.
+     * Print messages to inform user that the task has been added.
+     * @param taskList TaskList that contains all tasks at hand.
+     * @param ui Ui that deals with interaction with the user.
+     * @param storage Storage that deals with storing TaskList into hard drive.
```
Should there be an empty line between the description and parameters?

maxxng commented on src/main/java/duke/Duke.java:
```diff
@@ -1,3 +1,10 @@
+package duke;
+
+import duke.ui.*;
+import duke.task.*;
```
Should you try to avoid using wildcards when importing

maxxng commented on src/main/java/Duke.java:
```diff
@@ -1,11 +1,24 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileWriter;
```
Should you have an ordering of your import statements?

maxxng commented on src/main/java/Task.java:
```diff
@@ -2,20 +2,26 @@
         this.done = false;
     }
 
+    Task(String name, boolean done) {
```
Should your boolean variable name be isDone?

maxxng commented on src/main/java/Duke.java:
```diff
@@ -19,79 +32,145 @@ public static void main(String[] args) {
+    public static void writeTasks(File destination) throws IOException {
+        StringBuilder textToAdd = new StringBuilder();
+        FileWriter fw = new FileWriter(destination);
+        for (Task curr : currentTasks) {
```
Should your variable name curr be currentTask?

maxxng reviewed (commented):
In general, the code is easy to read and most variable names follow the coding convention/ is easy to understand.

xiinweii98 committed:
Added java Varargs to FindCommand

xiinweii98 committed:
Merge branch 'branch-A-Varargs'

xiinweii98 committed:
Tidied up codes to comply with new GUI standards and interactions.

-Removed excess and unused codes.
-Changed Duke and User images.
-Made sure responses from Duke is correct.
-Added introduction message from Duke on startup.

xiinweii98 committed:
Added assertions to mark done and delete command.

-Makes sure that input number is within range and valid.

xiinweii98 committed:
Added A-CodeQuality

Tried to find code to improve quality based on textbook.
-Edited Storage.update

xiinweii98 committed:
Added A-Lambdas

Tried adding lambda expressions.
-Edited TaskList.findTasks to make use of lambda expression instead of for-loop.

xiinweii98 committed:
CI maybe?

xiinweii98 committed:
Added B-Snooze

-Edited TaskList to keep track of additional ArrayList of snoozed tasks
-Added SnoozeCommand to snooze a task
-Edited ListCommand to be able to show snoozed commands
-Edited format of savedTasks on hard drive to cater to snoozed tasks
-Edited Storage class to cater to snoozed ArrayList
-Edited Parser class to cater to SnoozeCommand
-Edited Ui to cater to printing that is snooze related

-Added unSnooze method in TaskList but still unused. Will need to create an UnSnoozeCommand

xiinweii98 committed:
Merge pull request #1 from xiinweii98/branch-A-Assertions

Added assertions to mark done and delete command.

xiinweii98 committed:
Merge branch 'master' into branch-A-CodeQuality

xiinweii98 committed:
Merge pull request #2 from xiinweii98/branch-A-CodeQuality

Branch a code quality

xiinweii98 committed:
Merge branch 'master' into branch-A-Lambdas

xiinweii98 committed:
Merge pull request #3 from xiinweii98/branch-A-Lambdas

Branch a lambdas

xiinweii98 committed:
Merge branch 'master' into branch-A-CI

xiinweii98 committed:
Merge pull request #4 from xiinweii98/branch-A-CI

Branch a ci

xiinweii98 committed:
Merge branch 'master' into branch-BCD-Extension

# Conflicts:
#	data/savedTasks.txt
#	src/main/java/DeleteCommand.java

xiinweii98 committed:
Merge pull request #5 from xiinweii98/branch-BCD-Extension

Branch bcd extension

xiinweii98 committed:
Made better GUI.

xiinweii98 committed:
Updated user guide.

xiinweii98 committed:
Set theme jekyll-theme-slate
